tracing-core/src/dispatcher.rs

```diff
@@ -357,6 +357,59 @@ where
         .unwrap_or_else(|_| f(&Dispatch::none()))
 }
 
+/// Executes a closure with a reference to this thread's current [dispatcher].
+///
+/// Note that calls to `get_default` should not be nested; if this function is
+/// called while inside of another `get_default`, that closure will be provided
+/// with `Dispatch::none` rather than the previously set dispatcher.
+///
+/// [dispatcher]: ../dispatcher/struct.Dispatch.html
+#[cfg(feature = "std")]
+#[doc(hidden)]
```

Comment:
Adding this was necessary since constructing the `ValueSet` in a closure means we needed a `FnOnce` so that values can be moved into the closure (which happens primarily when users wrap values in `field::display`/`field::debug` without borrowing). Not supporting this would break at least some code. I opted to add a new function that takes a `FnOnce`, rather than capturing an `Option<FnOnce>` into the `FnMut` and `take`ing it, because that seemed significantly less efficient.

 We could probably commit to making this a public API (and even deprecate `get_default`, the version that takes a `FnMut`). This is _probably_ a more useful API than `get_default` taking a `FnMut`... I didn't make it public here, though, because introducing a new public API seemed like a job for another PR. 

Also, in 0.2, the `Value` changes will make `Value::debug` and `Value::display` also borrow rather than move. So, the `FnOnce` will no longer be necessary, and we could, alternatively, remove this function in 0.2, rather than deprecating the existing `get_current`.

So, there should probably be an actual design discussion... :)